# Titanic Dataset: Descriptive Statistics Worksheet

**Student Name:** Kaylah Benton

**Student Id:** 003049986

In [1]:
# Import Libraries
import numpy as np
import pandas as pd

In [2]:
# Load the Dataset
df = pd.read_csv('Titanic_Dataset.csv')
df

PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked  
0        0         A/5 21171   7.2500   NaN        S  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
4        0            373450   8.0500   NaN        S  
..     ...               ...      ...   ...      ...  
886      0            211536  13.0000   NaN        S  
887      0            112053  30.0000   B42        S  
888      2        W./C. 6607  23.4500   NaN        S  
889      0            111369  30.0000  C148        C  
890      0            370376   7.7500   NaN        Q  

[891 rows x 12 columns]

In [3]:
#Percentage of missing values
missing_data = df.isnull().sum()/len(df)*100 
print(missing_data)
df_new=df.copy()

PassengerId     0.000000
Survived        0.000000
Pclass          0.000000
Name            0.000000
Sex             0.000000
Age            19.865320
SibSp           0.000000
Parch           0.000000
Ticket          0.000000
Fare            0.000000
Cabin          77.104377
Embarked        0.224467
dtype: float64


In [4]:
df_new['Pclass']=df_new['Pclass'].astype('category')
df_new['Pclass']=df_new['Pclass'].map ({1:'First', 2: 'Second', 3: 'Third'})
# Fill missing Age values based on the median age for each Pclass and Sex group
df_new['Age'] = df_new.groupby(['Pclass', 'Sex'],observed=True)['Age'].transform(lambda x: x.fillna(x.median()))
df_filtered = df_new.copy()
df_filtered.drop(columns=['Embarked','Cabin'],inplace=True)
print("\n Data Frame after dropping categorical column embarked & Cabin: \n")
df_filtered


 Data Frame after dropping categorical column embarked & Cabin: 



PassengerId  Survived  Pclass  \
0              1         0   Third   
1              2         1   First   
2              3         1   Third   
3              4         1   First   
4              5         0   Third   
..           ...       ...     ...   
886          887         0  Second   
887          888         1   First   
888          889         0   Third   
889          890         1   First   
890          891         0   Third   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female  21.5      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare  
0        0         A/5 21171   7.2500  
1        0          PC 17599  71.2833  
2        0  STON/O2. 3101282   7.9250  
3        0            113803  53.1000  
4        0            373450   8.0500  
..     ...               ...      ...  
886      0            211536  13.0000  
887      0            112053  30.0000  
888      2        W./C. 6607  23.4500  
889      0            111369  30.0000  
890      0            370376   7.7500  

[891 rows x 10 columns]

In [5]:
def classify_age(age):
    if pd.isnull(age):
        return 'Unknown'
    elif age < 13:
        return 'Child'
    elif age < 18:
        return 'Teen'
    elif age < 35:
        return 'Young Adult'
    elif age < 60:
        return 'Adult'
    else:
        return 'Senior'
df_filtered['AgeGroup'] = df_filtered['Age'].apply(classify_age)

df_filtered['FamilySize'] = df_filtered['SibSp'] + df_filtered['Parch'] + 1
#Adds up siblings/spouses and parents/children on board, plus 1 for the passenger.

df_filtered['IsAlone'] = (df_filtered['FamilySize'] == 1).astype(int)
#Creates a binary flag — 1 if the passenger had no family aboard, otherwise 0.

df_filtered

PassengerId  Survived  Pclass  \
0              1         0   Third   
1              2         1   First   
2              3         1   Third   
3              4         1   First   
4              5         0   Third   
..           ...       ...     ...   
886          887         0  Second   
887          888         1   First   
888          889         0   Third   
889          890         1   First   
890          891         0   Third   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female  21.5      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare     AgeGroup  FamilySize  IsAlone  
0        0         A/5 21171   7.2500  Young Adult           2        0  
1        0          PC 17599  71.2833        Adult           2        0  
2        0  STON/O2. 3101282   7.9250  Young Adult           1        1  
3        0            113803  53.1000        Adult           2        0  
4        0            373450   8.0500        Adult           1        1  
..     ...               ...      ...          ...         ...      ...  
886      0            211536  13.0000  Young Adult           1        1  
887      0            112053  30.0000  Young Adult           1        1  
888      2        W./C. 6607  23.4500  Young Adult           4        0  
889      0            111369  30.0000  Young Adult           1        1  
890      0            370376   7.7500  Young Adult           1        1  

[891 rows x 13 columns]

### Part 1: Summary Statistics

---
**Q1.** Display summary statistics for all numeric columns.

In [6]:
# Write your code here

df_filtered.describe()

PassengerId    Survived         Age       SibSp       Parch  \
count   891.000000  891.000000  891.000000  891.000000  891.000000   
mean    446.000000    0.383838   29.112424    0.523008    0.381594   
std     257.353842    0.486592   13.304424    1.102743    0.806057   
min       1.000000    0.000000    0.420000    0.000000    0.000000   
25%     223.500000    0.000000   21.500000    0.000000    0.000000   
50%     446.000000    0.000000   26.000000    0.000000    0.000000   
75%     668.500000    1.000000   36.000000    1.000000    0.000000   
max     891.000000    1.000000   80.000000    8.000000    6.000000   

             Fare  FamilySize     IsAlone  
count  891.000000  891.000000  891.000000  
mean    32.204208    1.904602    0.602694  
std     49.693429    1.613459    0.489615  
min      0.000000    1.000000    0.000000  
25%      7.910400    1.000000    0.000000  
50%     14.454200    1.000000    1.000000  
75%     31.000000    2.000000    1.000000  
max    512.329200   11.000000    1.000000

**a.** What is the average age of passengers? <br>
**b.** What is the median fare paid?<br>
**c.** What is the maximum number of siblings/spouses (`SibSp`) a passenger had aboard?<br>
**d.** Comment on the range of the `Fare` column (min to max). Any outliers?


**a. Average age of passengers:**  
The average (mean) age of passengers was **29 years**.

**b. Median fare paid:**  
The median fare paid was **$14.45**.

**c. Maximum number of siblings/spouses aboard (SibSp):**  
The maximum number of siblings or spouses a passenger had aboard was **8**.

**d. Range of the Fare column (min to max):**  
The fare ranged from **$0.00 to $512.33**.  
This wide range suggests the presence of **outliers**, as a few passengers paid significantly higher fares than most others.


### Part 2: Survival Rate Analysis

---
**Q2.** Calculate the survival rate by gender.

In [7]:
# Write your code here
survival_rate_by_gender = df_filtered.groupby("Sex")["Survived"].mean()

print(survival_rate_by_gender)

Sex
female    0.742038
male      0.188908
Name: Survived, dtype: float64


**a.** What is the survival rate for males?<br>
**b.** What is the survival rate for females?<br>
**c.** What might explain this difference?


**a. Survival rate for males:**  
The survival rate for **males** was **0.1889**, or approximately **18.9%**.

**b. Survival rate for females:**  
The survival rate for **females** was **0.7420**, or approximately **74.2%**.

**c. Explanation of the difference:**  
The large difference in survival rates reflects the **“women and children first”** evacuation policy during the Titanic disaster.  
Women were given priority for lifeboat spots, while many men, particularly those in lower passenger classes, had fewer opportunities to survive.  
This outcome highlights how social norms and lifeboat access significantly influenced survival chances.


---
**Q3.** Calculate the survival rate by passenger class.

In [8]:
# Write your code here
print(df_filtered.groupby("Pclass", observed=True)["Survived"].sum()/len(df_filtered)*100)

Pclass
First     15.263749
Second     9.764310
Third     13.355780
Name: Survived, dtype: float64


**a.** Which class had the highest survival rate? <br>
**b.** Which class had the lowest?<br>
**c.** What might this tell you about access to lifeboats or crew priority?


**a. Which class had the highest survival rate?**  
The **1st Class** had the highest survival rate (**15.26%** of all passengers survived from this class).

**b. Which class had the lowest survival rate?**  
The **2nd Class** had the lowest survival rate (**9.76%** of all passengers survived from this class).

**c. What might this tell you about access to lifeboats or crew priority?**  
These percentages suggest that **first-class passengers had better access to lifeboats** and were likely prioritized by the crew.  
Second-class passengers had the lowest survival proportion, indicating **delayed access or lower priority**, while third-class passengers, though more numerous, had slightly higher survival than second class, possibly because some were lucky or closer to certain exits.  
Overall, **passenger class influenced survival outcomes**, reflecting **social and spatial hierarchies on the Titanic**.




### Part 3: Correlation Analysis

---
**Q4.** Create a correlation matrix for the following columns: <br>
Age, Fare, SibSp, Parch and Survived

In [9]:
# Write your code here
correlation_matrix = df_filtered[["Age", "Fare", "SibSp", "Parch", "Survived"]].corr()
print(correlation_matrix)

               Age      Fare     SibSp     Parch  Survived
Age       1.000000  0.122692 -0.249854 -0.175526 -0.059579
Fare      0.122692  1.000000  0.159651  0.216225  0.257307
SibSp    -0.249854  0.159651  1.000000  0.414838 -0.035322
Parch    -0.175526  0.216225  0.414838  1.000000  0.081629
Survived -0.059579  0.257307 -0.035322  0.081629  1.000000


**a.** Which variable has the strongest positive correlation with `Survived`?<br>
**b.** Does age have a strong effect on survival? Explain using the correlation value.<br>
**c.** Based on this matrix, which variables do you think will help in a prediction model?


**a. Which variable has the strongest positive correlation with `Survived`?**  
The variable **Fare** has the strongest positive correlation with survival (**r = 0.2573**).  
This indicates that passengers who paid higher fares (often in higher classes) were more likely to survive.

**b. Does age have a strong effect on survival? Explain using the correlation value.**  
No, **Age** has a very weak negative correlation with survival (**r = -0.0596**), suggesting that age had **little effect** on whether a passenger survived.

**c. Based on this matrix, which variables do you think will help in a prediction model?**  
Variables likely to help predict survival include:  
- **Fare**, since it shows the strongest positive correlation.  
- **Parch** (number of parents/children aboard) may add slight predictive value.   

Overall, **Fare**, **Passenger Class (Pclass)**, and **Sex** (not in this matrix) are typically the most important features in a Titanic survival prediction model.


---
**Q5.** Based on your findings, describe a group of passengers that was most likely to survive.
Use variables like `Sex`, `Pclass`, and `Fare` to support your answer.

Based on the data from `df_filtered`:

- **Sex:** Females were far more likely to survive than males, with a survival rate of **74.2%** compared to **18.9%** for males.  
- **Passenger Class (Pclass):** Interestingly, when looking at survival as a percentage of total passengers:
  - **First Class:** 15.26% of all passengers survived  
  - **Second Class:** 9.76% survived  
  - **Third Class:** 13.36% survived  
  This shows that **first-class passengers still had the highest proportion of survivors**, while second-class passengers had the lowest. Third-class survival, while lower than first class, was higher than second class in this calculation.  
- **Fare:** Passengers paying higher fares generally had a higher likelihood of survival, consistent with the positive correlation between **Fare** and **Survived** (**r = 0.257**).

**Conclusion:**  
The group most likely to survive were **females traveling in first class who paid higher fares**.  
This highlights how **gender, socioeconomic status, and fare** influenced survival chances, reflecting **crew priority and access to lifeboats** during the Titanic disaster.
